R/ScenarioConfiguration.R

```diff
@@ -0,0 +1,181 @@
+    #' to the simulation
+    setTestParameters = function(value) {
+      if (missing(value)) {
+        private$.setTestParameters
+      } else {
+        ospsuite.utils::validateIsLogical(value)
+        private$.setTestParameters <- value
+      }
+    },
+    #' @field simulateSteadyState Boolean representing whether the simulation will be brought to a steady-state first
+    simulateSteadyState = function(value) {
+      if (missing(value)) {
+        private$.simulateSteadyState
+      } else {
+        ospsuite.utils::validateIsLogical(value)
+        private$.simulateSteadyState <- value
+      }
+    },
+    #' @field simulationTime Boolean representing whether the simulation will be brought to a steady-state first
+    simulationTime = function(value) {
+      if (missing(value)) {
+        private$.simulationTime
+      } else {
+        ospsuite.utils::validateIsNumeric(value)
+        if (value < 0) {
+          stop(paste0("simulationTime must be a positive numerical value, but the value is ", value))
+        }
+        private$.simulationTime <- value
+      }
+    },
+    #' @field pointsPerMinute Resultion of the ouputs in points per minute
+    pointsPerMinute = function(value) {
+      if (missing(value)) {
+        private$.pointsPerMinute
+      } else {
+        ospsuite.utils::validateIsNumeric(value)
+        if (value < 0) {
+          stop(paste0("pointsPerMinute must be a positive numerical value, but the value is ", value))
+        }
+        private$.pointsPerMinute <- value
+      }
+    },
+    #' @field steadyStateTime Time in minutes to simulate if simulating steady-state. May be NULL
+    steadyStateTime = function(value) {
+      if (missing(value)) {
+        private$.steadyStateTime
+      } else {
+        ospsuite.utils::validateIsNumeric(value)
+        if (value < 0) {
+          stop(paste0("steadyStateTime must be a positive numerical value, but the value is ", value))
+        }
+        private$.steadyStateTime <- value
+      }
+    },
+    #' @field paramSheets Names of the sheets from the parameters-excel file that will be applied to the simulation
+    paramSheets = function(value) {
+      if (missing(value)) {
+        private$.paramSheets
+      } else {
+        stop("Field paramSheets is read-only! Use functions 'addParamSheet' and 'removeParamSheet' to add or remove
+a parameter sheet from the list")
+      }
+    },
+#' @field simulationType Type of the simulation - "Individual" or "Population". If "Population", population characteristics
+#' are created based on information stored in \code{populationParamsFile}.
+#' Default is "Individual"
+simulationType = function(value) {
+  if (missing(value)) {
+    private$.simulationType
+  } else {
+    if (value %in% c("Individual", "Population")) {
+      private$.simulationType <- value
+    } else {
+      stop("Wrong value for 'simulationType'! Accepted values are 'Individual
+                and 'Population'")
```

Comment:
I think this should be printed to a single line:

``` r
stop("Wrong value for 'simulationType'! Accepted values are 'Individual
                and 'Population'")
#> Error: Wrong value for 'simulationType'! Accepted values are 'Individual
#>                 and 'Population'
```

<sup>Created on 2022-04-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>